src/set_theory/ordinal/natural_ops.lean

```diff
@@ -0,0 +1,325 @@
+/-
+Copyright (c) 2022 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+
+import set_theory.ordinal.arithmetic
+
+/-!
+# Natural operations on ordinals
+
+The goal of this file is to define natural addition and multiplication on ordinals, and provide a
+basic API. The natural addition of two ordinals `a ♯ b` is recursively defined as the least ordinal
+greater than `a' ♯ b` and `a ♯ b'` for `a' < a` and `b' < b`. The natural multiplication `a ⨳ b` is
+likewise recursively defined as the least ordinal such that `a ⨳ b ♯ a' ⨳ b'` is greater than
+`a' ⨳ b ♯ a ⨳ b'` for any `a' < a` and `b' < b`.
+
+These operations form a rich algebraic structure: they're commutative, associative, preserve order,
+have the usual `0` and `1` from ordinals, and distribute over one another.
+
+Moreover, these operations are the addition and multiplication of ordinals when viewed as
+combinatorial `game`s. This makes them particularly useful for game theory.
+
+Finally, both operations admit simple, intuitive descriptions in terms of the Cantor normal form.
+The natural addition of two ordinals corresponds to adding their Cantor normal forms as if they were
+polynomials in `ω`. Likewise, their natural multiplication corresponds to multiplying the Cantor
+normal forms as polynomials.
+
+# Implementation notes
+
+Given the rich algebraic structure of these two operations, we choose to create a type synonym
+`nat_ordinal`, where we provide the appropriate instances. However, to avoid casting back and forth
+between both types, we attempt to prove and state most results on `ordinal`.
+
+# Todo
+
+- Define natural multiplication and provide a basic API.
+- Prove the characterizations of natural addition and multiplication in terms of the Cantor normal
+  form.
+-/
+
+universes u v
+
+open function
+
+noncomputable theory
+
+/-- A type synonym for ordinals with natural addition and multiplication. -/
+def nat_ordinal : Type* := ordinal
```

Comment:
I'm hesitant about the name `nat_` for this, it's already used for the naturals in Lean. Are there any other possible names?

Reply:
I'm not aware of any alternate name for natural operations. `natural_ordinal` seems too wordy to me. Is `ordinal_nat` any better?

Reply:
I don't think so, but I also don't think the name is problematic enough to be annoying

Reply:
An alternate name for the natural sum (and the name I first saw it called) is the Hessenberg sum: https://en.wikipedia.org/wiki/Ordinal_arithmetic#Natural_operations

Reply:
I've opened #14546 to add a docstring with these names.